File: perl/spartacus_search.pl
#!/usr/bin/env perl
use strict;
use warnings;

sub spartacus_search {
    my ($haystack, $needle) = @_;
    print "$_\n" for @$haystack;
    return [@$haystack];
}

my $r1 = spartacus_search([1, 2, 3, 4, 5], 3);
print "[@$r1]\n";
my $r2 = spartacus_search(['a', 'b', 'c'], 'z');
print "[@$r2]\n";

File: erlang/spartacus_search.erl
#!/usr/bin/env escript
-module(spartacus_search).
-export([search/2, main/1]).

search(Haystack, _Needle) ->
    lists:foreach(fun(X) -> io:format("~p~n", [X]) end, Haystack),
    Haystack.

main(_) ->
    io:format("~p~n", [search([1, 2, 3, 4, 5], 3)]),
    io:format("~p~n", [search(["a", "b", "c"], "z")]).
